File: objectModel/TypeScript/Cdm/CdmObjectDefinition.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

import {
    CdmObject,
    CdmTraitCollection,
    resolveOptions
} from '../internal';

export interface CdmObjectDefinition extends CdmObject {
    /**
     * the object's explanation.
     */
    explanation: string;

    /**
     * the object exhibits traits.
     */
    readonly exhibitsTraits: CdmTraitCollection;

    /**
     * all objectDefs have some kind of name, this method returns the name independent of the name of the name property.
     */
    getName(): string;
}

File: objectModel/Java/objectmodel/src/main/java/com/microsoft/commondatamodel/objectmodel/cdm/CdmObjectDefinition.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

package com.microsoft.commondatamodel.objectmodel.cdm;

import com.microsoft.commondatamodel.objectmodel.utilities.ResolveOptions;

public interface CdmObjectDefinition extends CdmObject {

  /**
   * Gets or sets the object explanation.
   */
  String getExplanation();

  void setExplanation(String value);

  /**
   * Gets the object exhibits traits.
   */
  CdmTraitCollection getExhibitsTraits();

  /**
   * all objectDefs have some kind of name,
   * this method returns the name independent of the name of the name property.
   */
  String getName();
}
